scripts/update_config.js

```diff
@@ -0,0 +1,143 @@
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+const fs = require('fs');
+const readline = require('readline');
+const program = require('commander');
+const extend = require('extend');
+const lisk = require('lisk-js');
+
+let oldConfigPath;
+let newConfigPath;
+
+program
+	.version('0.1.1')
+	.arguments('<old_config_file> <new_config_file>')
+	.action((oldConfig, newConfig) => {
+		oldConfigPath = oldConfig;
+		newConfigPath = newConfig;
+	})
+	.option('-p, --password <string>', 'Password for secret encryption', '')
+	.parse(process.argv);
+
+if (!oldConfigPath && !newConfigPath) {
+	console.error('error: no config file provided.');
+	process.exit(1);
+}
+
+console.info('Starting configuration migration...');
+const oldConfig = JSON.parse(fs.readFileSync(oldConfigPath, 'utf8'));
+const newConfig = JSON.parse(fs.readFileSync(newConfigPath, 'utf8'));
+
+// Values to keep from new config file
+delete oldConfig.version;
+delete oldConfig.minVersion;
+
+// Rename old port to new wsPort
+oldConfig.httpPort = oldConfig.port;
+oldConfig.wsPort = oldConfig.httpPort + 1;
+delete oldConfig.port;
+
+oldConfig.db.max = oldConfig.db.poolSize;
+delete oldConfig.db.poolSize;
+
+delete oldConfig.peers.options.limits;
+
+oldConfig.transactions.maxTransactionsPerQueue =
+	oldConfig.transactions.maxTxsPerQueue;
+delete oldConfig.transactions.maxTxsPerQueue;
+
+delete oldConfig.loading.verifyOnLoading;
+delete oldConfig.dapp;
+
+// Peers migration
+oldConfig.peers.list.map(p => {
+	p.wsPort = p.port;
+	delete p.port;
+});
+
+if (oldConfig.forging.secret && oldConfig.forging.secret.length) {
+	if (!program.password.trim()) {
+		const rl = readline.createInterface({
+			input: process.stdin,
+			output: process.stdout,
+		});
+		rl.question(
+			'We found some secrets in your config, if you want to migrate, please type in your password (enter to skip): ',
+			password => {
+				rl.close();
+				migrateSecrets(password);
+				copyTheConfigFile();
+			}
+		);
+		// To Patch the password support
+		rl.stdoutMuted = true;
+		rl._writeToOutput = function _writeToOutput(stringToWrite) {
+			if (rl.stdoutMuted) rl.output.write('*');
+			else rl.output.write(stringToWrite);
+		};
+	} else {
+		migrateSecrets(program.password);
+		copyTheConfigFile();
+	}
+} else {
+	migrateSecrets('');
+	copyTheConfigFile();
+}
+
+function migrateSecrets(password) {
+	oldConfig.forging.delegates = [];
+
+	if (!password.trim()) {
+		console.info('\nSkipping the secret migration.');
+		delete oldConfig.forging.secret;
+		return;
+	}
+
+	console.info('\nMigrating your secrets...');
+	oldConfig.forging.defaultPassword = password;
+	oldConfig.forging.secret.map(secret => {
```

Comment:
Either you want a `forEach` here or you want to assign `oldConfig.forging.delegates` to the result of the `map` function, and return the object rather than pushing it.